### DISCUSSION TITLE
Disable interaction autoplay

### DISCUSSION THREAD

**Discussion by liaoyeguo** [storybookjs/storybook#25011] [Help]
### Summary

I add a play function to my story for testing, interactions are auto playing when user visit stories or doc. How can I disable this behavior and trigger interaction manually

**Comment by jonniebigodes** [comment]
@liaoyeguo, thanks for reaching out to us with your question. Appreciate it 🙏. What you're asking about is a feature that's been asked often [here](https://github.com/storybookjs/storybook/discussions/23616) and also [here](https://github.com/storybookjs/storybook/discussions/24785). However, for part of your question (i.e., docs), you could configure it not to run automatically. See [here](https://storybook.js.org/docs/api/doc-block-story) for documentation on configuring it.


Looking forward to hearing from you. 

Have a fantastic day.

Stay safe

**Reply by KevinGhadyani-Okta** [reply]
None of those links solve the problem in my case. I'm not the OP, but I'm someone who's negatively affected by autoplaying interaction tests.

The docs page doesn't matter because interaction tests are disabled by default for that page. What matters are the stories pages.

1. I design a story to be a certain way, and I don't want interaction tests to be changing that.
2. I only want interaction tests to run if I'm on the Interactions tab, not when doing a11y checks.
3. When doing a11y checks, it specifically breaks where something fades-in. The a11y plugin sees the changing colors and assumes the contrast is wrong but doesn't re-run after the transition is complete. This wouldn't be an issue if the interaction test run on page load, but they do.
 
It only errors _sometimes_. That's the biggest problem. Interaction tests running cause a race condition. I'd prefer these interaction tests don't run until I tell them to. Even better if I can set one to autoplay while testing in Storybook's UI rather than having it always running; messing up my stories.

## Not a solution
A solution was proposed in one of those links to separate stories with interaction tests into another directory, but a11y is running on all stories. It doesn't exclude some just because they have interactions enabled.

This also requires doubling up on stories just to write interaction tests where I could've just used Jest or Playwright directly instead. What advantage are Interaction tests even bringing to Storybook? If you wanna open a `Select` dropdown on page load, then rename this to "Storybook pre-load" and don't advertise it as an alternative to running unit tests.

**Comment by John0x** [comment]
Having the same issue.
The autoplay should exist for non docs as well, preferably on a global level. I don't want the interaction tests to automatically run upon opening the story in storybook.
@jonniebigodes

**Comment by kerematam** [comment]
Autoplay messes snapshot tests. Please support this feature.

**Reply by MrSuddenJoy** [reply]
Same here. Should be a toggle somewhere (`Settings` tab maybe?)

**Comment by dlarroder** [comment]
@jonniebigodes are there any plans in future releases to support disabling interaction tests autoplay?

**Comment by thobas-dnvgl** [comment]
We've been using Storybook for a long time but without the interactions/tests. We just started writing our first test and what a disappoint to see that it runs by default when opening a story and that there is NO WAY to disable this behavior.

**Comment by danfincher** [comment]
You can setup this behavior yourself. If you have a [testRunnerConfig.ts](https://storybook.js.org/docs/writing-tests/integrations/test-runner), you can add this to preVisit:

```typescript
// Inject flag so play functions do not run in browser
await page.evaluate(() => {
    (window as any).__isTestRunner = true;
});
```

Then in your play function, check for that condition:

```typescript
play: async () => {
    if ((window as any).__isTestRunner) {
        // play function only runs during test mode not in browser
    }
}.
```

**Reply by MrSuddenJoy** [reply]
@danfincher thank you so much. This is proper way of hadling this.

**Reply by benface** [reply]
I don't think this works with the new Vitest addon. :(

**Comment by nicooprat** [comment]
I think it's solved by migrating interaction tests to the Vitest addon, where tests can be run manually & independently from the UI: https://storybook.js.org/docs/writing-tests/integrations/vitest-addon/migration-guide

<img width="490" height="252" alt="image" src="https://github.com/user-attachments/assets/d4be514d-20a7-47dd-b0fe-049b63137257" />

See this note: https://storybook.js.org/docs/writing-tests/integrations/test-runner

> The test runner has been superseded by the [Vitest addon](https://storybook.js.org/docs/writing-tests/integrations/vitest-addon), which offers the same functionality, powered by the faster and more modern [Vitest](https://vitest.dev/) browser mode. It also enables the full Storybook Test experience, allowing you to run interaction, accessibility, and visual tests from your Storybook app.

So I guess those play functions are considered deprecated.

**Reply by benface** [reply]
I don't think this is true. How are we supposed to write interaction tests with the Vitest addon other than with `play` functions? [The documentation page for the addon](https://storybook.js.org/docs/writing-tests/integrations/vitest-addon#how-it-works) even mentions:

> Stories are tested in two ways: a smoke test to ensure it renders and, if a [play function](https://storybook.js.org/docs/writing-tests/interaction-testing#writing-interaction-tests) is defined, that function is run and any [assertions made](https://storybook.js.org/docs/writing-tests/interaction-testing#asserting-with-expect) within it are validated.
